src/pip/_internal/commands/configuration.py

```diff
@@ -237,13 +241,32 @@ def open_in_editor(self, options, args):
             raise PipError("Could not determine appropriate file.")
 
         try:
-            subprocess.check_call([editor, fname])
+            if PY3:
```

Comment:
I suggest moving all the not-subprocess-call stuff into `_determine_editor` and then, changing the original subprocess call line to:

```py
subprocess.check_call(editor_args + [fname])
```